src/command/backup/backup.c

```diff
@@ -399,14 +398,18 @@ backupBuildIncrPrior(const InfoBackup *infoBackup)
                     "last backup label = %s, version = %s", strPtr(manifestData(result)->backupLabel),
                     strPtr(manifestData(result)->backrestVersion));
 
-                // Warn if compress option changed
-                if (cfgOptionBool(cfgOptCompress) != manifestPriorData->backupOptionCompress)
+                // Warn if compress-type option changed
+                if (compressTypeEnum(cfgOptionStr(cfgOptCompressType)) != manifestPriorData->backupOptionCompressType)
                 {
                     LOG_WARN_FMT(
-                        "%s backup cannot alter compress option to '%s', reset to value in %s",
-                        strPtr(cfgOptionStr(cfgOptType)), cvtBoolToConstZ(cfgOptionBool(cfgOptCompress)),
-                        strPtr(backupLabelPrior));
-                    cfgOptionSet(cfgOptCompress, cfgSourceParam, VARBOOL(manifestPriorData->backupOptionCompress));
+                        "%s backup cannot alter " CFGOPT_COMPRESS_TYPE " option to '%s', reset to value in %s",
+                        strPtr(cfgOptionStr(cfgOptType)),
+                        strPtr(compressTypeStr(compressTypeEnum(cfgOptionStr(cfgOptCompressType)))), strPtr(backupLabelPrior));
+
+                    // Set the compression type back to whatever was in the prior backup.  This is not strictly needed since we
+                    // could store compression type on a per file basis, but it seems simplest and safest for now.
+                    cfgOptionSet(
+                        cfgOptCompressType, cfgSourceParam, VARSTR(compressTypeStr(manifestPriorData->backupOptionCompressType)));
```

Comment:
This isn't a new issue, but shouldn't we also reset the compress-level? If I have a prior FULL backup (compress defaults), then run an INCR backup with --no-compress, I get the WARN that it is being reset to the FULL backup and the INCR manifest winds up with:

option-compress=true
**option-compress-level=0**
option-compress-level-network=3
option-compress-type="gz"

whereas the FULL backup had the default compress-level=6. The INCR has compress-level of 0 vs the default of 6 because I passed --no-compress which must have changed the compress-level to 0.

Reply:
Yeah, that's a problem.  I think it is OK for the user to set a new compress-level as long as the type does not change, but if they disable compression we should reset the value to whatever was used before.

It's a bit complicated because we did not always store the compress-level option in the manifest (though it has been there for a pretty long time), so we'll need to set to default when not present or just throw an error.

Reply:
It might be a little more complicated than "if they disable compression". See email: subject 'Compression-level issue'.

Reply:
Well, the way the code works, it's really "if the compress type changes", not  "was compression disabled" so I think it will work in either case.

Reply:
I did try to write a test for this, but since backupBuildIncrPrior() currently only raises warnings it turned into a real mess.  To test this effectively we'd need to change where we call backupBuildIncrPrior().  So, for now just use check and add a big comment, as discussed.